modules/apps/fragment/fragment-impl/src/main/java/com/liferay/fragment/internal/util/configuration/FragmentEntryConfigurationParserImpl.java

```diff
@@ -412,6 +413,9 @@ private String _getColorPickerCssVariable(String fieldValue) {
 
 		FrontendTokenDefinition frontendTokenDefinition =
 			_frontendTokenDefinitionRegistry.getFrontendTokenDefinition(
+				layoutSet.getCompanyId(),
+				CETUtil.getLayoutSetThemeCSSCETExternalReferenceCode(
+					layoutSet.getLayoutSetId()),
```

Comment:
This is not related to this PR, but I _believe_ this won't be ok if one sets a ThemeCSS CET for a particular page. I don't know if setting a frontend-token-definition specific to a page was possible before FTDs were part of ThemeCSSCET. But now that they are, the FTD can be different on a page-level basis.

Please see my comment about how `ClientExtensionsServicePreAction` [computes the applicable ThemeCSSCET for a layout](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/client-extension/client-extension-service/src/main/java/com/liferay/client/extension/internal/events/ClientExtensionsServicePreAction.java#L174-L201), I think that logic is relevant here.

Reply:
Hey @dsanz you're correct. This is problematic when setting a theme or themeCSS CX at the page level. We decided this would be out of scope for this epic because changing this behavior would also require rethinking how a user creates stylebooks through the admin ui. Currently, you can only create a stylebook based on the frontend-token-definition.json file that is associated with the public pages of a site. In order to be able to properly handle this at the page level, you would need to be able to create a stylebook for any frontend-token-definition.json file that exists in the instance, provided by either a theme or themeCSS CX. I think we should improve this, but in a separate epic.